subsys/bluetooth/mesh/net.c

```diff
@@ -773,17 +576,11 @@ static void bt_mesh_net_local(struct k_work *work)
 static void net_tx_cred_get(struct bt_mesh_net_tx *tx, uint8_t *nid,
 			    const uint8_t **enc, const uint8_t **priv)
 {
-	if (!IS_ENABLED(CONFIG_BT_MESH_LOW_POWER) || !tx->friend_cred) {
-		tx->friend_cred = 0U;
-		*nid = tx->sub->keys[tx->sub->kr_flag].nid;
-		*enc = tx->sub->keys[tx->sub->kr_flag].enc;
-		*priv = tx->sub->keys[tx->sub->kr_flag].privacy;
-		return;
-	}
-
-	if (friend_cred_get(tx->sub, BT_MESH_ADDR_UNASSIGNED, nid, enc, priv)) {
-		BT_WARN("Falling back to master credentials");
-
+	if (tx->friend_cred &&
+		!friend_cred_find(tx->sub, BT_MESH_ADDR_UNASSIGNED,
+				nid, enc, priv)) {
+			return;
+	} else {
```

Comment:
Actually, I don't think this is a bug. The Friend node only sends two messages through the network layer (clear and clear confirm), and both are encrypted with the master security credentials. I think this should be reversed.

Reply:
Well, According Mesh Profile Spec 1.0.1 Section 4.2.2.4 Publish Friendship Credentials Flag

When Publish Friendship Credential Flag is set to 1 and the friendship security material is not available, the master security material shall be used, otherwise, friendship security shell be used.

Reply:
Yes, but this was taken care of by the second if, no?

Reply:
I'm remove second commit from this pr, due to this credential_flag cannot be set to 1 since the node does not support low power feature as defined by spec `4.4.1.2.7 Model Publication state`.

Actually, this not bug here, but will be bug in mod_pub_set.